stage1/init/common/pod.go

```diff
@@ -623,12 +747,9 @@ func PodToSystemd(p *stage1commontypes.Pod, interactive bool, flavor string, pri
 	return nil
 }
 
-// evaluateAppMountPath tries to resolve symlinks within the path.
-// It returns the actual relative path for the given path.
-// TODO(yifan): This is a temporary fix for systemd-nspawn not handling symlink mounts well.
-// Could be removed when https://github.com/systemd/systemd/issues/2860 is resolved, and systemd
-// version is bumped.
-func evaluateAppMountPath(appRootfs, path string) (string, error) {
+// evaluateSymlinksInsideApp tries to resolve symlinks within the path.
+// It returns the actual path relative to the pod for the given path.
+func evaluateSymlinksInsideApp(appRootfs, path string) (string, error) {
 	link := appRootfs
 
 	paths := strings.Split(path, "/")
```

Comment:
Hm... not really, `SplitList` splits things according to `ListSeparator`, which is ":" on Unix.

One thing I could do is using `os.PathSeparator` although since we're dealing with symlinks, this is very Unixy anyways.